### ISSUE TITLE
Server crashes due to the std.experimenta.allocator assertion

### ISSUE THREAD

**Issue by tchaloupka** [rejectedsoftware/vibe.d#1704]
I'm getting this error when the server crashes. Not sure what the problem might be yet as it happens randomly (same request works just fine) and did not happened previously (with no changes on the server side).

Might be related to https://issues.dlang.org/show_bug.cgi?id=15637

vibe-d-0.8.0-beta.3
dmd v2.073.1
Ubuntu 16.04.2 LTS x86_64

```
[7C14BD32:00000000 2017.03.06 09:31:33.420 FATAL] Worker thread terminated due to uncaught error: Assertion failure
[7C14BD32:00000000 2017.03.06 09:31:33.420 dbg] Full error: core.exception.AssertError@std/experimental/allocator/building_blocks/region.d(232): Assertion failure
----------------
??:? _d_assert [0x9d1d6c]
??:? void std.experimental.allocator.building_blocks.region.__assert(int) [0xa20aa6]
/usr/include/dmd/phobos/std/experimental/allocator/building_blocks/region.d:232 pure nothrow @nogc bool std.experimental.allocator.building_blocks.region.__T6RegionTS3std12experimental9allocator12gc_allocator11GCAllocatorVki16VE3std8typec
ons43__T4FlagVAyaa13_67726f77446f776e7761726473Z4Flagi0Z.Region.expand(ref void[], ulong) [0x90a932]
/usr/include/dmd/phobos/std/experimental/allocator/building_blocks/stats_collector.d:368 pure nothrow @nogc bool std.experimental.allocator.building_blocks.stats_collector.__T14StatsCollectorTS3std12experimental9allocator15building_blocks
6region140__T6RegionTS3std12experimental9allocator12gc_allocator11GCAllocatorVki16VE3std8typecons43__T4FlagVAyaa13_67726f77446f776e7761726473Z4Flagi0Z6RegionVmi1024Vmi0Z.StatsCollector.expandImpl!(null, 0).expandImpl(ref void[], ulong) [0
x8f26f8]
/usr/include/dmd/phobos/std/experimental/allocator/building_blocks/stats_collector.d:348 pure nothrow @nogc bool std.experimental.allocator.building_blocks.stats_collector.__T14StatsCollectorTS3std12experimental9allocator15building_blocks
6region140__T6RegionTS3std12experimental9allocator12gc_allocator11GCAllocatorVki16VE3std8typecons43__T4FlagVAyaa13_67726f77446f776e7761726473Z4Flagi0Z6RegionVmi1024Vmi0Z.StatsCollector.expand(ref void[], ulong) [0x8f1fc3]
/usr/include/dmd/phobos/std/experimental/allocator/building_blocks/allocator_list.d:386 pure nothrow @nogc bool std.experimental.allocator.building_blocks.allocator_list.AllocatorList!(std.experimental.allocator.building_blocks.allocator_
list.__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ
.Factory, std.experimental.allocator.building_blocks.null_allocator.NullAllocator).AllocatorList.expand(ref void[], ulong) [0x8f16ad]
/usr/include/dmd/phobos/std/experimental/allocator/common.d:345 nothrow bool std.experimental.allocator.common.reallocate!(std.experimental.allocator.building_blocks.allocator_list.AllocatorList!(std.experimental.allocator.building_blocks
.allocator_list.__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13Nul
lAllocatorZ.Factory, std.experimental.allocator.building_blocks.null_allocator.NullAllocator).AllocatorList).reallocate(ref std.experimental.allocator.building_blocks.allocator_list.AllocatorList!(std.experimental.allocator.building_block
s.allocator_list.__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13Nu
llAllocatorZ.Factory, std.experimental.allocator.building_blocks.null_allocator.NullAllocator).AllocatorList, ref void[], ulong) [0x90b7bd]
/usr/include/dmd/phobos/std/experimental/allocator/package.d:1788 bool std.experimental.allocator.__T14CAllocatorImplTS3std12experimental9allocator15building_blocks14allocator_list404__T13AllocatorListTS3std12experimental9allocator15build
ing_blocks14allocator_list233__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_a
llocator13NullAllocatorZ7FactoryTS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ13AllocatorListVE3std8typecons32__T4FlagVAyaa8_696e646972656374Z4Flagi0Z.CAllocatorImpl.reallocate(ref void[], ulong) [0x90afee
]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/utils/vibe/utils/array.d:92 @trusted void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.reserve(ulong) [0x803413]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/utils/vibe/utils/array.d:178 @safe void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.grow(ulong) [0x803946]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/utils/vibe/utils/array.d:114 @safe void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.put(ubyte[]) [0x8036c1]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/utils/vibe/utils/array.d:123 @trusted void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.put(const(ubyte[])) [0x8037af]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/stream/vibe/stream/operations.d:394 @safe void vibe.stream.operations.readUntilSmall!(vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readUntilSmall(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, const(ubyte[]), ulong) [0x900fbe]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/stream/vibe/stream/operations.d:142 @safe void vibe.stream.operations.readUntil!(vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readUntil(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, const(ubyte[]), ulong) [0x900a6e]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/stream/vibe/stream/operations.d:56 @safe void vibe.stream.operations.readLine!(vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readLine(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, ulong, immutable(char)[]) [0x9009cb]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/stream/vibe/stream/operations.d:41 @safe ubyte[] vibe.stream.operations.readLine!(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readLine(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ulong, immutable(char)[], std.experimental.allocator.IAllocator) [0x900919]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/inet/vibe/inet/message.d:52 @safe immutable(char)[] vibe.inet.message.parseRFC5322Header!(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).parseRFC5322Header(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.dictionarylist.DictionaryList!(immutable(char)[], false, 12uL, false).DictionaryList, ulong, std.experimental.allocator.IAllocator, bool).readStringLine() [0x901ec9]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/inet/vibe/inet/message.d:57 @safe void vibe.inet.message.parseRFC5322Header!(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).parseRFC5322Header(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.dictionarylist.DictionaryList!(immutable(char)[], false, 12uL, false).DictionaryList, ulong, std.experimental.allocator.IAllocator, bool) [0x901b4c]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/http/vibe/http/server.d:2095 @safe void vibe.http.server.parseRequestHeader!(vibe.core.stream.InputStream).parseRequestHeader(vibe.http.server.HTTPServerRequest, vibe.core.stream.InputStream, std.experimental.allocator.IAllocator, ulong) [0x8f6332]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/http/vibe/http/server.d:1868 @safe bool vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool, scope std.experimental.allocator.IAllocator) [0x8d8100]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/http/vibe/http/server.d:1757 @trusted void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo).__lambda4() [0x8d7a37]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/http/vibe/http/server.d:1753 @safe void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo) [0x8d78a9]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/http/vibe/http/server.d:1628 nothrow @safe void vibe.http.server.listenHTTPPlain(vibe.http.server.HTTPServerSettings, vibe.http.server.HTTPServerContext).doListen(vibe.http.server.HTTPListenInfo, vibe.http.server.HTTPServerContext, bool, bool).__lambda6(vibe.core.net.TCPConnection) [0x8d7084]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/core/vibe/core/drivers/libevent2_tcp.d:618 void vibe.core.drivers.libevent2_tcp.ClientTask.execute() [0x9aaf85]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/core/vibe/core/core.d:609 void vibe.core.core.makeTaskFuncInfo!(void delegate()).makeTaskFuncInfo(ref void delegate()).callDelegate(vibe.core.core.TaskFuncInfo*) [0x95344c]
../../../../../.dub/packages/vibe-d-0.8.0-beta.3/vibe-d/core/vibe/core/core.d:1250 void vibe.core.core.CoreTask.run() [0x94e926]
??:? void core.thread.Fiber.run() [0xa3103b]
??:? fiber_entryPoint [0xa30d9e]
```

**Comment by tchaloupka** [comment]
Closer to the problem cause.

It can be reproduced with simple app:
```D
import std.stdio;
import vibe.d;

shared static this() {
	auto settings = new HTTPServerSettings();
	settings.port = 8080;
	settings.tlsContext = createTLSContext(TLSContextKind.server);
	settings.tlsContext.useCertificateChainFile("server.crt");
	settings.tlsContext.usePrivateKeyFile("server.key");

	auto router = new URLRouter;
	router.get("*", &handler);

	listenHTTP(settings, router);
}

void handler(scope HTTPServerRequest req, scope HTTPServerResponse res) {
	res.writeBody("hello world");
}
```

It has to be https (can't reproduce it with http)

And then with `curl -ik "https://localhost:8080" -H"Authorization: Bearer aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"`

(it seems to be a problem which appears with some request size)

It crash with:

```
[618A9F0B:618B430B trc] --------------------
[618A9F0B:618B430B trc] HTTP server request:
[618A9F0B:618B430B trc] --------------------
[618A9F0B:618B430B trc] GET / HTTP/1.1
[618A9F0B:618B430B CRITICAL] CoreTaskFiber was terminated unexpectedly: Assertion failure
[618A9F0B:618B430B dia] Full error: core.exception.AssertError@std/experimental/allocator/building_blocks/region.d(232): Assertion failure
[618A9F0B:618B430B dia] ----------------
[618A9F0B:618B430B dia] ??:? _d_assert [0x9dd494]
[618A9F0B:618B430B dia] ??:? void std.experimental.allocator.building_blocks.region.__assert(int) [0xa27b0a]
[618A9F0B:618B430B dia] /usr/include/dmd/phobos/std/experimental/allocator/building_blocks/region.d:232 pure nothrow @nogc bool std.experimental.allocator.building_blocks.region.__T6RegionTS3std12experimental9allocator12gc_allocator11GCAllocatorVki16VE3std8typecons43__T4FlagVAyaa13_67726f77446f776e7761726473Z4Flagi0Z.Region.expand(ref void[], ulong) [0x90ec92]
[618A9F0B:618B430B dia] /usr/include/dmd/phobos/std/experimental/allocator/building_blocks/stats_collector.d:368 pure nothrow @nogc bool std.experimental.allocator.building_blocks.stats_collector.__T14StatsCollectorTS3std12experimental9allocator15building_blocks6region140__T6RegionTS3std12experimental9allocator12gc_allocator11GCAllocatorVki16VE3std8typecons43__T4FlagVAyaa13_67726f77446f776e7761726473Z4Flagi0Z6RegionVmi1024Vmi0Z.StatsCollector.expandImpl!(null, 0).expandImpl(ref void[], ulong) [0x8f5e04]
[618A9F0B:618B430B dia] /usr/include/dmd/phobos/std/experimental/allocator/building_blocks/stats_collector.d:348 pure nothrow @nogc bool std.experimental.allocator.building_blocks.stats_collector.__T14StatsCollectorTS3std12experimental9allocator15building_blocks6region140__T6RegionTS3std12experimental9allocator12gc_allocator11GCAllocatorVki16VE3std8typecons43__T4FlagVAyaa13_67726f77446f776e7761726473Z4Flagi0Z6RegionVmi1024Vmi0Z.StatsCollector.expand(ref void[], ulong) [0x8f56cf]
[618A9F0B:618B430B dia] /usr/include/dmd/phobos/std/experimental/allocator/building_blocks/allocator_list.d:386 pure nothrow @nogc bool std.experimental.allocator.building_blocks.allocator_list.AllocatorList!(std.experimental.allocator.building_blocks.allocator_list.__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ.Factory, std.experimental.allocator.building_blocks.null_allocator.NullAllocator).AllocatorList.expand(ref void[], ulong) [0x8f4db9]
[618A9F0B:618B430B dia] /usr/include/dmd/phobos/std/experimental/allocator/common.d:345 nothrow bool std.experimental.allocator.common.reallocate!(std.experimental.allocator.building_blocks.allocator_list.AllocatorList!(std.experimental.allocator.building_blocks.allocator_list.__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ.Factory, std.experimental.allocator.building_blocks.null_allocator.NullAllocator).AllocatorList).reallocate(ref std.experimental.allocator.building_blocks.allocator_list.AllocatorList!(std.experimental.allocator.building_blocks.allocator_list.__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ.Factory, std.experimental.allocator.building_blocks.null_allocator.NullAllocator).AllocatorList, ref void[], ulong) [0x90fb31]
[618A9F0B:618B430B dia] /usr/include/dmd/phobos/std/experimental/allocator/package.d:1788 bool std.experimental.allocator.__T14CAllocatorImplTS3std12experimental9allocator15building_blocks14allocator_list404__T13AllocatorListTS3std12experimental9allocator15building_blocks14allocator_list233__T13AllocatorListS1324vibe4http6server20handleHTTPConnectionFNfC4vibe4core3net13TCPConnectionC4vibe4http6server14HTTPListenInfoZ9__lambda4MFNeZ9__lambda1TS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ7FactoryTS3std12experimental9allocator15building_blocks14null_allocator13NullAllocatorZ13AllocatorListVE3std8typecons32__T4FlagVAyaa8_696e646972656374Z4Flagi0Z.CAllocatorImpl.reallocate(ref void[], ulong) [0x90f346]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/utils/vibe/utils/array.d:92 @trusted void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.reserve(ulong) [0x82cd13]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/utils/vibe/utils/array.d:178 @safe void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.grow(ulong) [0x82d246]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/utils/vibe/utils/array.d:114 @safe void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.put(ubyte[]) [0x82cfc1]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/utils/vibe/utils/array.d:123 @trusted void vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender.put(const(ubyte[])) [0x82d0af]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/stream/vibe/stream/operations.d:394 @safe void vibe.stream.operations.readUntilSmall!(vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readUntilSmall(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, const(ubyte[]), ulong) [0x905f38]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/stream/vibe/stream/operations.d:142 @safe void vibe.stream.operations.readUntil!(vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readUntil(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, const(ubyte[]), ulong) [0x9059f2]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/stream/vibe/stream/operations.d:56 @safe void vibe.stream.operations.readLine!(vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readLine(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.array.AllocAppender!(ubyte[], ubyte).AllocAppender, ulong, immutable(char)[]) [0x90594f]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/stream/vibe/stream/operations.d:41 @safe ubyte[] vibe.stream.operations.readLine!(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).readLine(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ulong, immutable(char)[], std.experimental.allocator.IAllocator) [0x90589d]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/inet/vibe/inet/message.d:52 @safe immutable(char)[] vibe.inet.message.parseRFC5322Header!(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).parseRFC5322Header(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.dictionarylist.DictionaryList!(immutable(char)[], false, 12uL, false).DictionaryList, ulong, std.experimental.allocator.IAllocator, bool).readStringLine() [0x906e4d]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/inet/vibe/inet/message.d:57 @safe void vibe.inet.message.parseRFC5322Header!(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef).parseRFC5322Header(vibe.internal.freelistref.__T11FreeListRefTC4vibe4http6server22LimitedHTTPInputStreamVbi1Z.FreeListRef, ref vibe.utils.dictionarylist.DictionaryList!(immutable(char)[], false, 12uL, false).DictionaryList, ulong, std.experimental.allocator.IAllocator, bool) [0x906ad0]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/http/vibe/http/server.d:2109 @safe void vibe.http.server.parseRequestHeader!(vibe.core.stream.InputStream).parseRequestHeader(vibe.http.server.HTTPServerRequest, vibe.core.stream.InputStream, std.experimental.allocator.IAllocator, ulong) [0x8f9a98]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/http/vibe/http/server.d:1882 @safe bool vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool, scope std.experimental.allocator.IAllocator) [0x8df2a4]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/http/vibe/http/server.d:1771 @trusted void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo).__lambda4() [0x8debb3]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/http/vibe/http/server.d:1767 @safe void vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo) [0x8dea25]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/http/vibe/http/server.d:1642 nothrow @safe void vibe.http.server.listenHTTPPlain(vibe.http.server.HTTPServerSettings, vibe.http.server.HTTPServerContext).doListen(vibe.http.server.HTTPListenInfo, vibe.http.server.HTTPServerContext, bool, bool).__lambda6(vibe.core.net.TCPConnection) [0x8de200]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/core/vibe/core/drivers/libevent2_tcp.d:618 void vibe.core.drivers.libevent2_tcp.ClientTask.execute() [0x9af96d]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/core/vibe/core/core.d:620 void vibe.core.core.makeTaskFuncInfo!(void delegate()).makeTaskFuncInfo(ref void delegate()).callDelegate(vibe.core.core.TaskFuncInfo*) [0x955f04]
[618A9F0B:618B430B dia] ../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/core/vibe/core/core.d:1269 void vibe.core.core.CoreTask.run() [0x9510d2]
[618A9F0B:618B430B dia] ??:? void core.thread.Fiber.run() [0xa397f7]
[618A9F0B:618B430B dia] ??:? fiber_entryPoint [0xa3955a]
[618A9F0B:618B430B dia] ??:? [0xffffffff]
[618A9F0B:618B430B dia] Main thread exiting
Program exited with code 255
```

Tested with multiple dmd versions (also with dmd-2.073.2).
vibe-d-0.8.0-beta.4
Fedora 25, Ubuntu 16.04.2 - x86_64

dub.json:
```
{
	"name": "issue1704",
	"dependencies": {
		"vibe-d": "~>0.8.0-beta"
	},
	"versions": ["VibeDefaultMain"]
}
```

**Comment by s-ludwig** [comment]
Will be fixed by #1760, which targets a memory leak that was also introduced by using AllocatorList/RegionAllocator from std.experimental. But based on the call sequence up to that `reallocate()` I agree that it's very likely to be issue 15637.
